plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTableBuilder.java

```diff
@@ -64,53 +68,62 @@ public static DynamicTable buildFromPql(PinotMetadata pinotMetadata, SchemaTable
-                    if (columnHandle == null) {
-                        throw new ColumnNotFoundException(schemaTableName, sortItem.getColumn());
-                    }
-                    orderByBuilder.add(new OrderByExpression(columnHandle.getColumnName(), sortItem.isIsAsc()));
-                }
-                orderBy = orderByBuilder.build();
+        if (queryContext.getOrderByExpressions() != null) {
+            ImmutableList.Builder<OrderByExpression> orderByBuilder = ImmutableList.builder();
+            for (OrderByExpressionContext orderByExpressionContext : queryContext.getOrderByExpressions()) {
+                ExpressionContext expressionContext = orderByExpressionContext.getExpression();
+                checkState(expressionContext.getType() == ExpressionContext.Type.IDENTIFIER, "Unexpected order by expression: '%s'", expressionContext);
+                String name = expressionContext.getIdentifier();
+                PinotColumnHandle columnHandle = getColumnHandle(schemaTableName, name, columnHandles);
+                orderByBuilder.add(new OrderByExpression(columnHandle.getColumnName(), orderByExpressionContext.isAsc()));
             }
+            orderBy = orderByBuilder.build();
         }
 
-        List<String> groupByColumns;
-        if (request.getGroupBy() == null) {
-            groupByColumns = ImmutableList.of();
-        }
-        else {
-            groupByColumns = resolvePinotColumns(schemaTableName, request.getGroupBy().getExpressions(), columnHandles);
+        Optional<String> filter = Optional.empty();
+        if (pinotQuery.getFilterExpression() != null) {
+            String formatted = formatFilter(schemaTableName, queryContext.getFilter(), columnHandles);
+            filter = Optional.of(formatted);
         }
 
-        Optional<String> filter;
-        if (request.getFilterQuery() != null) {
-            filter = Optional.of(convertFilter(request.getPinotQuery(), columnHandles));
-        }
-        else {
-            filter = Optional.empty();
-        }
-        QueryContext queryContext = BrokerRequestToQueryContextConverter.convert(request);
+        List<String> selectionColumns = ImmutableList.of();
+        List<String> groupByColumns = ImmutableList.of();
+
         ImmutableList.Builder<PinotColumnHandle> aggregateColumnsBuilder = ImmutableList.builder();
-        if (request.getAggregationsInfo() != null) {
+        if (queryContext.getAggregationFunctions() != null) {
+            checkState(queryContext.getAggregationFunctions().length > 0, "Aggregation Functions is empty");
             for (AggregationFunction aggregationFunction : queryContext.getAggregationFunctions()) {
                 aggregateColumnsBuilder.add(new PinotColumnHandle(
                         aggregationFunction.getResultColumnName(),
                         toTrinoType(aggregationFunction.getFinalResultColumnType())));
             }
+            if (queryContext.getGroupByExpressions() != null) {
```

Comment:
Why is this nested inside the aggregations block?
I think it's possible for queries to have GROUP BY without aggregations.

`SELECT 1 FROM t GROUP BY c`?

Reply:
reverted.

Reply:
Let's add a test with this query:

1. Ensure that the plan contains a TableScan (so that we are sure that the connector participated in the query and the engine didn't do it all itself).
2. The results are correct.

Reply:
I created https://github.com/trinodb/trino/issues/9167 to test this in BaseConnectorTest as well.